### PR TITLE
giac: makedepend on mpfr-devel and pari-devel

### PR DESCRIPTION
**Pull request by tornaria**
- mpfr is used for arbitrary precision floats, otherwise they are
   limited to double precision (53 bits ~ 13 digits)

 - pari is used for number theory functions

In addition to the two new makedepends, there is a small patch that
fixes one line of test output to match a change in pari.

With this commit, make check goes from 12 tests failed to 1 test failed.

See #28989.

### CODE REVIEW

**Comment by dkwo** [comment]
I can confirm it builds and works fine on `x86_64-musl`.

**Comment by tornaria** [comment]
> I can confirm it builds and works fine on `x86_64-musl`.

Did you run make check? Because there are some test failures that only seem to happen for x86_64-musl, e.g.:

```
0>> integrate(atan(sqrt(1-x**2)),x,0,1)
"Unable to handle singularities of -1/2*pi-sqrt(2)*(-1/2*pi-atan(sqrt(2)*x*((-1/2*(-2*sqrt(-x^2+1)+2)/x)^2-1)/(-2*sqrt(-x^2+1)+2)))-atan(x*((-1/2*(-2*sqrt(-x^2+1)+2)/x)^2-1)/(-2*sqrt(-x^2+1)+2))+x*atan(sqrt(-x^2+1)) at [undef] Error: Bad Argument Value"
```

I don't use musl myself.

**Comment by ericonr** [comment]
> 2021-02-28T22:50:27.3688440Z FAIL: chk_xavier

That's on glibc. It might be nice to report the musl errors if they can be traced back to musl :) I will try to group them together.

**Comment by dkwo** [comment]
It actually fails 8 of 30 tests on musl:

FAIL: chk_integrate
FAIL: chk_morley_demo
FAIL: chk_xavier
FAIL: chk_fhan0
FAIL: chk_fhan14
FAIL: chk_fhan16
FAIL: chk_fhan18
FAIL: chk_fhan7

Can provide more details if needed.

**Comment by ericonr** [comment]
I am aware of the test failures on musl, but we'd have to track down where they go wrong.

For now, it would be nice if it at least worked on glibc cleanly :)

**Comment by ericonr** [comment]
Remember to squash the commits :)

I don't really understand much about math libraries... If the resulting package works, we can at least leave a comment in the template that some tests need fixing.

**Comment by tornaria** [comment]
I think this is ready to merge.

All tests pass now, both glibc and musl.

Squashed in one commit (https://github.com/void-linux/void-packages/commit/06ed2689c4fa3b3ad4a14c64e6b75ccff0234e90)

File: srcpkgs/giac/template [hunk 1]
```diff
@@ -14,6 +14,11 @@ homepage="https://www-fourier.ujf-grenoble.fr/~parisse/giac.html"
 distfiles="https://www-fourier.ujf-grenoble.fr/~parisse/debian/dists/stable/main/source/giac_${version%.*}-${version##*.}.tar.gz"
 checksum=bbd6f0aafe373de3c9ed53d21878224e847359fdd9a7cdf9e9096e7c960f10b4
 
+if [ "$XBPS_TARGET_LIBC" = musl ]; then
+	# need more than 4*65536 stack, see try_parse() in gen.cc line 11812
+	LDFLAGS+=" -Wl,-z,stack-size=1048576"
```
Comment (ericonr) [inline srcpkgs/giac/template:19]: ah, nice find! You can apply this unconditionally, it shouldn't affect the resulting glibc binary.

File: srcpkgs/giac/patches/invmod.patch [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+@@ -92,7 +92,10 @@
+   gen invenv(const gen & g,environment * env){
+     if (g.type==_USER)
+       return g._USERptr->inv();
+-    return invmod(g,env->modulo);
++    if (env->moduloon)
++      return invmod(g,env->modulo);
++    else
++      return gen(1)/g;
+   }
```
Comment (ericonr) [inline srcpkgs/giac/patches/invmod.patch:12]: Is this patch from upstream as well?

Comment (tornaria) [reply srcpkgs/giac/patches/invmod.patch:12]: It's my patch. I'll send it upstream.

File: srcpkgs/giac/template (hunk 1 above)
Comment (tornaria) [reply srcpkgs/giac/template:19]: I was in doubt. Should I make the stack as small as required to pass tests (~270K seems enough)?

I'd rather give it some slack to be safe.

Comment (ericonr) [reply srcpkgs/giac/template:19]: Oh, I meant the test for musl wasn't necessary. And you should just set `LDFLAGS`, not add to it. The value is good, though 2MB is closer to glibc's default thread stack size, if you want to go with that.

**Comment by ericonr** [comment]
Wonderful, thanks for the detailed commit message

**Pull request merged**